src/wallet/wallet.cpp

```diff
@@ -3049,6 +3049,9 @@ unsigned int CWallet::GetKeyPoolSize() const
 
 bool CWallet::TopUpKeyPool(unsigned int kpSize)
 {
+    if (IsWalletFlagSet(WALLET_FLAG_BLANK_WALLET) || IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
```

Comment:
In commit "Key pool: Make TopUp fail if unexpected wallet flags are set" (b12121d52f46ac7e8455479bc9ded100cfccc841)

There should be a code comment here to explain what's mentioned in the PR description: that the blank wallet check is needed for TopUp to return false instead of true in a wallet without any keyman objects. It would also be good to say why that's important, because on the surface it seems like a bad choice to return failure when nothing failed.

Relatedly, I would love to be able to drop this commit if it isn't actually necessary or if there isn't a specific example of a bug you could imagine this check preventing. Wallet flag code is confusing and just referencing flags in random places for no apparent reason and with no explanation isn't the greatest thing.

It's one thing to add sanity checks in asserts or CHECK_NONFATAL calls, but something else to add sanity checks in real code without clear "this condition is never expected to happen" comments so people reading the code don't waste time trying to imagine cases where it could actually do anything.

Reply:
Looking at this change further, I don't think it's needed, so I've removed this commit.